Add script for conversion to BIDS #8

Kyota-exe opened the pull request:
This script converts the existing data for MGH and MNI scans from the previous structure/format to BIDS.
![Screenshot 2023-12-27 130852](https://github.com/spinal-cord-7t/coil-qc-code/assets/56048818/3aeb6214-7cbc-4210-a523-279fc659db4a)
In order to use it, first download the archives from the OSF repository, unzip them and rename them to `SITE-original`. Then run the `bids_conversion.py` script from this directory. It will generate a folder called `outputs`, which has the following directory structure:
```
├───MGH
│   ├───Spinoza6
│   │   ├───anat
│   │   └───fmap
│   ├───SubD
│   │   ├───anat
│   │   └───fmap
│   ├───SubL
│   │   ├───anat
│   │   └───fmap
│   └───SubR
│       ├───anat
│       └───fmap
└───MNI
    ├───Spinoza6
    │   ├───anat
    │   └───fmap
    ├───SubD
    │   ├───anat
    │   └───fmap
    ├───SubL
    │   ├───anat
    │   └───fmap
    └───SubR
        ├───anat
        └───fmap
```
This is an example of the `tree` output for `MGH/SubD`:
```
├───anat
│       SubD_gre-uncombined-cRX01-ph_T2starw.json
│       SubD_gre-uncombined-cRX01-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX01_T2starw.json
│       SubD_gre-uncombined-cRX01_T2starw.nii.gz
│       SubD_gre-uncombined-cRX02-ph_T2starw.json
│       SubD_gre-uncombined-cRX02-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX02_T2starw.json
│       SubD_gre-uncombined-cRX02_T2starw.nii.gz
│       SubD_gre-uncombined-cRX03-ph_T2starw.json
│       SubD_gre-uncombined-cRX03-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX03_T2starw.json
│       SubD_gre-uncombined-cRX03_T2starw.nii.gz
│       SubD_gre-uncombined-cRX04-ph_T2starw.json
│       SubD_gre-uncombined-cRX04-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX04_T2starw.json
│       SubD_gre-uncombined-cRX04_T2starw.nii.gz
│       SubD_gre-uncombined-cRX05-ph_T2starw.json
│       SubD_gre-uncombined-cRX05-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX05_T2starw.json
│       SubD_gre-uncombined-cRX05_T2starw.nii.gz
│       SubD_gre-uncombined-cRX06-ph_T2starw.json
│       SubD_gre-uncombined-cRX06-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX06_T2starw.json
│       SubD_gre-uncombined-cRX06_T2starw.nii.gz
│       SubD_gre-uncombined-cRX07-ph_T2starw.json
│       SubD_gre-uncombined-cRX07-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX07_T2starw.json
│       SubD_gre-uncombined-cRX07_T2starw.nii.gz
│       SubD_gre-uncombined-cRX08-ph_T2starw.json
│       SubD_gre-uncombined-cRX08-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX08_T2starw.json
│       SubD_gre-uncombined-cRX08_T2starw.nii.gz
│       SubD_gre-uncombined-cRX09-ph_T2starw.json
│       SubD_gre-uncombined-cRX09-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX09_T2starw.json
│       SubD_gre-uncombined-cRX09_T2starw.nii.gz
│       SubD_gre-uncombined-cRX10-ph_T2starw.json
│       SubD_gre-uncombined-cRX10-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX10_T2starw.json
│       SubD_gre-uncombined-cRX10_T2starw.nii.gz
│       SubD_gre-uncombined-cRX11-ph_T2starw.json
│       SubD_gre-uncombined-cRX11-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX11_T2starw.json
│       SubD_gre-uncombined-cRX11_T2starw.nii.gz
│       SubD_gre-uncombined-cRX12-ph_T2starw.json
│       SubD_gre-uncombined-cRX12-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX12_T2starw.json
│       SubD_gre-uncombined-cRX12_T2starw.nii.gz
│       SubD_gre-uncombined-cRX13-ph_T2starw.json
│       SubD_gre-uncombined-cRX13-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX13_T2starw.json
│       SubD_gre-uncombined-cRX13_T2starw.nii.gz
│       SubD_gre-uncombined-cRX14-ph_T2starw.json
│       SubD_gre-uncombined-cRX14-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX14_T2starw.json
│       SubD_gre-uncombined-cRX14_T2starw.nii.gz
│       SubD_gre-uncombined-cRX15-ph_T2starw.json
│       SubD_gre-uncombined-cRX15-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX15_T2starw.json
│       SubD_gre-uncombined-cRX15_T2starw.nii.gz
│       SubD_gre-uncombined-cRX16-ph_T2starw.json
│       SubD_gre-uncombined-cRX16-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX16_T2starw.json
│       SubD_gre-uncombined-cRX16_T2starw.nii.gz
│       SubD_gre-uncombined-cRX17-ph_T2starw.json
│       SubD_gre-uncombined-cRX17-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX17_T2starw.json
│       SubD_gre-uncombined-cRX17_T2starw.nii.gz
│       SubD_gre-uncombined-cRX18-ph_T2starw.json
│       SubD_gre-uncombined-cRX18-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX18_T2starw.json
│       SubD_gre-uncombined-cRX18_T2starw.nii.gz
│       SubD_gre-uncombined-cRX19-ph_T2starw.json
│       SubD_gre-uncombined-cRX19-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX19_T2starw.json
│       SubD_gre-uncombined-cRX19_T2starw.nii.gz
│       SubD_gre-uncombined-cRX20-ph_T2starw.json
│       SubD_gre-uncombined-cRX20-ph_T2starw.nii.gz
│       SubD_gre-uncombined-cRX20_T2starw.json
│       SubD_gre-uncombined-cRX20_T2starw.nii.gz
│       SubD_gre_combined-64_T2starw.json
│       SubD_gre_combined-64_T2starw.nii.gz
│       SubD_gre_combined-66_T2starw.json
│       SubD_gre_combined-66_T2starw.nii.gz
│       SubD_gre_combined-67_T2starw.json
│       SubD_gre_combined-67_T2starw.nii.gz
│       SubD_gre_combined-68_T2starw.json
│       SubD_gre_combined-68_T2starw.nii.gz
│       SubD_mp2rage-INV1.json
│       SubD_mp2rage-INV1.nii.gz
│       SubD_mp2rage-INV2.json
│       SubD_mp2rage-INV2.nii.gz
│       SubD_mp2rage-UNI.json
│       SubD_mp2rage-UNI.nii.gz
│
└───fmap
        SubD_coilqa-sag-large-SNR.json
        SubD_coilqa-sag-large-SNR.nii.gz
        SubD_coilqa-sag-small-GFactor.json
        SubD_coilqa-sag-small-GFactor.nii.gz
        SubD_coilqa-tra-GFactor.json
        SubD_coilqa-tra-GFactor.nii.gz
        SubD_dream-medium-066-e1-39.json
        SubD_dream-medium-066-e1-39.nii.gz
        SubD_dream-medium-066-e1-42.json
        SubD_dream-medium-066-e1-42.nii.gz
        SubD_dream-medium-066-e2-39.json
        SubD_dream-medium-066-e2-39.nii.gz
        SubD_dream-medium-066-e2-40.json
        SubD_dream-medium-066-e2-40.nii.gz
        SubD_dream-medium-066-e2-41.json
        SubD_dream-medium-066-e2-41.nii.gz
        SubD_dream-medium-066-e2-42.json
        SubD_dream-medium-066-e2-42.nii.gz
        SubD_dream-medium-066-e2-43.json
        SubD_dream-medium-066-e2-43.nii.gz
        SubD_dream-medium-066-e2-44.json
        SubD_dream-medium-066-e2-44.nii.gz
        SubD_dream-medium-e1-33.json
        SubD_dream-medium-e1-33.nii.gz
        SubD_dream-medium-e1-36.json
        SubD_dream-medium-e1-36.nii.gz
        SubD_dream-medium-e2-33.json
        SubD_dream-medium-e2-33.nii.gz
        SubD_dream-medium-e2-34.json
        SubD_dream-medium-e2-34.nii.gz
        SubD_dream-medium-e2-35.json
        SubD_dream-medium-e2-35.nii.gz
        SubD_dream-medium-e2-36.json
        SubD_dream-medium-e2-36.nii.gz
        SubD_dream-medium-e2-37.json
        SubD_dream-medium-e2-37.nii.gz
        SubD_dream-medium-e2-38.json
        SubD_dream-medium-e2-38.nii.gz
        SubD_dream-medium-hwlimit-e1-45.json
        SubD_dream-medium-hwlimit-e1-45.nii.gz
        SubD_dream-medium-hwlimit-e1-48.json
        SubD_dream-medium-hwlimit-e1-48.nii.gz
        SubD_dream-medium-hwlimit-e2-45.json
        SubD_dream-medium-hwlimit-e2-45.nii.gz
        SubD_dream-medium-hwlimit-e2-46.json
        SubD_dream-medium-hwlimit-e2-46.nii.gz
        SubD_dream-medium-hwlimit-e2-47.json
        SubD_dream-medium-hwlimit-e2-47.nii.gz
        SubD_dream-medium-hwlimit-e2-48.json
        SubD_dream-medium-hwlimit-e2-48.nii.gz
        SubD_dream-medium-hwlimit-e2-49.json
        SubD_dream-medium-hwlimit-e2-49.nii.gz
        SubD_dream-medium-hwlimit-e2-50.json
        SubD_dream-medium-hwlimit-e2-50.nii.gz
        SubD_tfl-FAmap.json
        SubD_tfl-FAmap.nii.gz
        SubD_tfl-mag.json
        SubD_tfl-mag.nii.gz
```
I followed the general instructions detailed in #8 to figure out in which directory (`fmap` or `anat`) to place each file, but did not know how to name every file. I thus named them so that it clearly states the relevant information that distinguishes each file, and added the series number when I wasn't sure what exactly differentiates them (for example, for the GRE combined scans or the DREAM scans). For the DREAM scans for which we were not sure which ones to include/exclude, I simply included all of them as a first step, but it will be easy to modify this as needed. Now that the script is written, it is very easy to rename these files correctly so they follow the BIDS convention.

I would appreciate guidance concerning what the correct naming convention should be for these files. I have uploaded the generated outputs from this first version of the script in [NeuroPoly/Projects/Spine7T_Zurich/Data/outputs_2023-12-27.zip](https://drive.google.com/file/d/1jz2PcQLY4HXbFfYv4HFBje6gzsqeKa5U/view?usp=sharing).

jcohenadad reviewed (commented):
Thank you for getting it started @Kyota-exe ! This is a good start. A few high-level comments:
- We need to use 'sub-' as subject prefix
- We need to create a single "flat" dataset that includes all the sites. That way we only need to do one import on OpenNeuro. Otherwise it will be treated as one dataset per site. Which means that the site can be encoded in the subject name. Example: `sub-MGHspinoza`, `sub-MNIr`
- coilqc, dream, etc. are not BIDS compliant fields. To specify the acquisition sequence, you can use `acq`. To distinguish between orientation, etc. you can add fields, without dash or underscord, but using Camel case. Example: `sub-MNId_acq-coilqaSagLarge_SNR.json`. Now, this is still _not_ BIDS compliant, because the suffix `SNR` does not exist. My suggestion is to list the suffixes that we need, and then ask the question on the [neurostars/BIDS forum](https://neurostars.org/tag/bids), referring to this issue https://github.com/spinal-cord-7t/coil-qc-code/issues/8
- about "SubD_gre-uncombined-cRX01-ph_T2starw.json". I don't think "gre" is an authorized field either. If you wish to describe the sequence, you should probably use `acq-`.

Kyota-exe commented:
1179b35 has "flattened" the output directory of the script (the generated BIDS database) so that all sites are included as separate versions of each subject in accordance with [option 2 described in this article](https://bids-specification.readthedocs.io/en/v1.1.2/05-longitudinal-and-multi-site-studies.html#:~:text=The%20simplest%20way%20of%20dealing,and%20everything%20should%20just%20work.). After having sorted all the NYU DICOMs, I've also added NYU to the list of sites that the script supports with 3986165. 

The following is now the list of directories that the script generates:
```
sub-MGHd/
sub-MGHl/
sub-MGHr/
sub-MGHspinoza6/
sub-MNId/
sub-MNIl/
sub-MNIr/
sub-MNIspinoza6/
sub-NYUd/
sub-NYUl/
sub-NYUr/
```

jcohenadad commented:
About https://github.com/spinal-cord-7t/coil-qc-code/pull/9#issuecomment-1903329289, BIDS multisite datasets typically use the terminology SITE<index> instead of SITE<letter> to distinguish between subjects. It could be e.g.:
~~~
sub-MGH1/
sub-MGH2/
sub-MGH3/
~~~

And the file `participants.tsv` can help us keep track of who is who

jcohenadad commented:
@Kyota-exe would you be able to do a tree of the produced dataset so I can review how it looks like?

Kyota-exe commented:
@jcohenadad Of course. Thank you for your feedback.
I updated the filenames so that they use the `acq-` field, but I was not sure whether info specific on the scan itself such as the channel number for GRE and the series number & "e version" (?) for DREAM scans should but put inside this field.
I would greatly appreciate any feedback.

The following is the directory structure:
```
├───sub-MGH1
│   ├───anat
│   └───fmap
├───sub-MGH2
│   ├───anat
│   └───fmap
├───sub-MGH3
│   ├───anat
│   └───fmap
├───sub-MGH4
│   ├───anat
│   └───fmap
├───sub-MNI1
│   ├───anat
│   └───fmap
├───sub-MNI2
│   ├───anat
│   └───fmap
├───sub-MNI3
│   ├───anat
│   └───fmap
├───sub-MNI4
│   ├───anat
│   └───fmap
├───sub-NYU1
│   ├───anat
│   └───fmap
├───sub-NYU2
│   ├───anat
│   └───fmap
└───sub-NYU3
    ├───anat
    └───fmap
```
The following are the files inside `sub-MGH1`:
```
├───sub-MGH1
│   ├───anat
│   │       sub-MGH1_acq-greUncombined01_T2starw.json
│   │       sub-MGH1_acq-greUncombined01_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined02_T2starw.json
│   │       sub-MGH1_acq-greUncombined02_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined03_T2starw.json
│   │       sub-MGH1_acq-greUncombined03_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined04_T2starw.json
│   │       sub-MGH1_acq-greUncombined04_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined05_T2starw.json
│   │       sub-MGH1_acq-greUncombined05_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined06_T2starw.json
│   │       sub-MGH1_acq-greUncombined06_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined07_T2starw.json
│   │       sub-MGH1_acq-greUncombined07_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined08_T2starw.json
│   │       sub-MGH1_acq-greUncombined08_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined09_T2starw.json
│   │       sub-MGH1_acq-greUncombined09_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined10_T2starw.json
│   │       sub-MGH1_acq-greUncombined10_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined11_T2starw.json
│   │       sub-MGH1_acq-greUncombined11_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined12_T2starw.json
│   │       sub-MGH1_acq-greUncombined12_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined13_T2starw.json
│   │       sub-MGH1_acq-greUncombined13_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined14_T2starw.json
│   │       sub-MGH1_acq-greUncombined14_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined15_T2starw.json
│   │       sub-MGH1_acq-greUncombined15_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined16_T2starw.json
│   │       sub-MGH1_acq-greUncombined16_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined17_T2starw.json
│   │       sub-MGH1_acq-greUncombined17_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined18_T2starw.json
│   │       sub-MGH1_acq-greUncombined18_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined19_T2starw.json
│   │       sub-MGH1_acq-greUncombined19_T2starw.nii.gz
│   │       sub-MGH1_acq-greUncombined20_T2starw.json
│   │       sub-MGH1_acq-greUncombined20_T2starw.nii.gz
│   │       sub-MGH1_acq-mp2rageINV1.json
│   │       sub-MGH1_acq-mp2rageINV1.nii.gz
│   │       sub-MGH1_acq-mp2rageINV2.json
│   │       sub-MGH1_acq-mp2rageINV2.nii.gz
│   │       sub-MGH1_acq-mp2rageUNI.json
│   │       sub-MGH1_acq-mp2rageUNI.nii.gz
│   │
│   └───fmap
│           sub-MGH1_acq-coilqaSagLarge_SNR.json
│           sub-MGH1_acq-coilqaSagLarge_SNR.nii.gz
│           sub-MGH1_acq-coilqaSagSmall_GFactor.json
│           sub-MGH1_acq-coilqaSagSmall_GFactor.nii.gz
│           sub-MGH1_acq-coilqaTra_GFactor.json
│           sub-MGH1_acq-coilqaTra_GFactor.nii.gz
│           sub-MGH1_acq-dreamMedium066_e1_39.json
│           sub-MGH1_acq-dreamMedium066_e1_39.nii.gz
│           sub-MGH1_acq-dreamMedium066_e1_42.json
│           sub-MGH1_acq-dreamMedium066_e1_42.nii.gz
│           sub-MGH1_acq-dreamMedium066_e2_39.json
│           sub-MGH1_acq-dreamMedium066_e2_39.nii.gz
│           sub-MGH1_acq-dreamMedium066_e2_40.json
│           sub-MGH1_acq-dreamMedium066_e2_40.nii.gz
│           sub-MGH1_acq-dreamMedium066_e2_41.json
│           sub-MGH1_acq-dreamMedium066_e2_41.nii.gz
│           sub-MGH1_acq-dreamMedium066_e2_42.json
│           sub-MGH1_acq-dreamMedium066_e2_42.nii.gz
│           sub-MGH1_acq-dreamMedium066_e2_43.json
│           sub-MGH1_acq-dreamMedium066_e2_43.nii.gz
│           sub-MGH1_acq-dreamMedium066_e2_44.json
│           sub-MGH1_acq-dreamMedium066_e2_44.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e1_45.json
│           sub-MGH1_acq-dreamMediumHwlimit_e1_45.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e1_48.json
│           sub-MGH1_acq-dreamMediumHwlimit_e1_48.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e2_45.json
│           sub-MGH1_acq-dreamMediumHwlimit_e2_45.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e2_46.json
│           sub-MGH1_acq-dreamMediumHwlimit_e2_46.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e2_47.json
│           sub-MGH1_acq-dreamMediumHwlimit_e2_47.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e2_48.json
│           sub-MGH1_acq-dreamMediumHwlimit_e2_48.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e2_49.json
│           sub-MGH1_acq-dreamMediumHwlimit_e2_49.nii.gz
│           sub-MGH1_acq-dreamMediumHwlimit_e2_50.json
│           sub-MGH1_acq-dreamMediumHwlimit_e2_50.nii.gz
│           sub-MGH1_acq-dreamMedium_e1_33.json
│           sub-MGH1_acq-dreamMedium_e1_33.nii.gz
│           sub-MGH1_acq-dreamMedium_e1_36.json
│           sub-MGH1_acq-dreamMedium_e1_36.nii.gz
│           sub-MGH1_acq-dreamMedium_e2_33.json
│           sub-MGH1_acq-dreamMedium_e2_33.nii.gz
│           sub-MGH1_acq-dreamMedium_e2_34.json
│           sub-MGH1_acq-dreamMedium_e2_34.nii.gz
│           sub-MGH1_acq-dreamMedium_e2_35.json
│           sub-MGH1_acq-dreamMedium_e2_35.nii.gz
│           sub-MGH1_acq-dreamMedium_e2_36.json
│           sub-MGH1_acq-dreamMedium_e2_36.nii.gz
│           sub-MGH1_acq-dreamMedium_e2_37.json
│           sub-MGH1_acq-dreamMedium_e2_37.nii.gz
│           sub-MGH1_acq-dreamMedium_e2_38.json
│           sub-MGH1_acq-dreamMedium_e2_38.nii.gz
│           sub-MGH1_acq-tfl_FAmap.json
│           sub-MGH1_acq-tfl_FAmap.nii.gz
│           sub-MGH1_acq-tfl_mag.json
│           sub-MGH1_acq-tfl_mag.nii.gz
```
And the following is the file contents of `participants.tsv`:
```
participant_id  species age     sex     pathology       institution     field
sub-MGH4        homo sapiens    n/a     n/a     HC      MGH     7T
sub-MGH1        homo sapiens    n/a     n/a     HC      MGH     7T
sub-MGH2        homo sapiens    n/a     n/a     HC      MGH     7T
sub-MGH3        homo sapiens    n/a     n/a     HC      MGH     7T
sub-MNI4        homo sapiens    n/a     n/a     HC      MNI     7T
sub-MNI1        homo sapiens    n/a     n/a     HC      MNI     7T
sub-MNI2        homo sapiens    n/a     n/a     HC      MNI     7T
sub-MNI3        homo sapiens    n/a     n/a     HC      MNI     7T
sub-NYU1        homo sapiens    n/a     n/a     HC      NYU     7T
sub-NYU2        homo sapiens    n/a     n/a     HC      NYU     7T
sub-NYU3        homo sapiens    n/a     n/a     HC      NYU     7T
```

jcohenadad commented:
Great job! 👏   We're almost there I think. Few comments:

- `sub-MGH1_acq-greUncombined01_T2starw.json` --> `sub-MGH1_acq-uncombined01_T2starw.json` (no need to indicate the sequence type, it's already in the suffix)
- `sub-MGH1_acq-mp2rageINV1.nii.gz` --> `sub-MGH1_inv-1_part-mag_MP2RAGE.nii.gz` (see the [BIDS definition](https://bids-specification.readthedocs.io/en/stable/appendices/qmri.html#inputs-are-file-collections))
- `sub-MGH1_acq-mp2rageINV2.nii.gz` --> `sub-MGH1_inv-2_part-mag_MP2RAGE.nii.gz`
- `sub-MGH1_acq-mp2rageINV1.json` --> `sub-MGH1_inv-1_MP2RAGE.json`
- `sub-MGH1_acq-mp2rageINV2.json` --> `sub-MGH1_inv-2_MP2RAGE.json`
- `sub-MGH1_acq-mp2rageUNI.nii.gz` --> `sub-MGH1_UNIT1.nii.gz` also see my note below

> [!Note]
> Normally the T1map and UNIT1 should go under the derivatives as per the [convention](https://bids-specification.readthedocs.io/en/stable/appendices/qmri.html#quantitative-maps-are-derivatives), but to simplify our processing scripts I think it is reasonable to put everything under the _source_ folder. Otherwise it might create confusion to have some of our input data under the derivatives folder.

- Don't you have a T1map as well from the MP2RAGE sequence? If should, call it `sub-MGH1_T1map.nii.gz`
- `sub-MGH1_acq-coilqaSagLarge*` --> `sub-MGH1_acq-coilQaSagLarge` (to respect lowerCamelCase)
- `sub-MGH1_acq-dreamMedium066_e1_39.nii.gz` --> `sub-MGH1_acq-dreamMedium066e1.nii.gz` "_39" is not an acceptable suffix, and I presume it refers to the scan number (@evaalonsoortiz can confirm). Instead, you should be describing what information content is associated with that file, eg: magnitude, phase, etc. See [BIDS definition on B1+ maps](https://bids-specification.readthedocs.io/en/stable/appendices/qmri.html#tb1tfl-and-tb1rfm-specific-notes), and our [example dataset on OpenNeuro](https://openneuro.org/datasets/ds004906/versions/2.4.0) (look for the TFL sequence). 
- `sub-MGH1_acq-tfl_FAmap.nii.gz` --> same comment as above, see our example for the TFL sequence

I have more comments, but right now there are too many changes to apply, and some of my additional comments might be fixed after you address the ones above.

evaalonsoortiz commented:
1. Such as the acquisition protocol is set up, there is no T1 map output by the MP2RAGE. Is there usually? It might be an option in the protocol menu ... The acquisition protocol for the MP2RAGE scan was setup by the Marseille group, as it is part of the qMRI project protocol. 

2. I believe *39* is indeed a scan number.

jcohenadad commented:
> Such as the acquisition protocol is set up, there is no T1 map output by the MP2RAGE. Is there usually? It might be an option in the protocol menu ... The acquisition protocol for the MP2RAGE scan was setup by the Marseille group, as it is part of the qMRI project protocol.

yes, it is the main point of the MP2RAGE: to get a T1map. We should have the Marseille folks chip into this conversation— do you want me to do it?

evaalonsoortiz commented:
Sure, let's double check with them. From what I've read in papers (ex: https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/full/10.1002/nbm.4142) it sounds like you have the generate the look up table yourself using the UNI1 images.

Kyota-exe commented:
I made some modifications to the filenames based on the feedback by @jcohenadad. The only modification I have not yet implemented is the proper naming of the DREAM files: the only thing I see clearly differentiating them is the series number, hence why I originally put it in the filename (if I removed it, I would have multiple of the same filename in the directory, which is not possible), but since this does not follow the BIDS convention, I will further investigate these scans.

Kyota-exe commented:
The following is the contents of `sub-MGH1` after the latest modifications:
```
├───anat
│       sub-MGH1_acq-uncombined01_T2starw.json
│       sub-MGH1_acq-uncombined01_T2starw.nii.gz
│       sub-MGH1_acq-uncombined02_T2starw.json
│       sub-MGH1_acq-uncombined02_T2starw.nii.gz
│       sub-MGH1_acq-uncombined03_T2starw.json
│       sub-MGH1_acq-uncombined03_T2starw.nii.gz
│       sub-MGH1_acq-uncombined04_T2starw.json
│       sub-MGH1_acq-uncombined04_T2starw.nii.gz
│       sub-MGH1_acq-uncombined05_T2starw.json
│       sub-MGH1_acq-uncombined05_T2starw.nii.gz
│       sub-MGH1_acq-uncombined06_T2starw.json
│       sub-MGH1_acq-uncombined06_T2starw.nii.gz
│       sub-MGH1_acq-uncombined07_T2starw.json
│       sub-MGH1_acq-uncombined07_T2starw.nii.gz
│       sub-MGH1_acq-uncombined08_T2starw.json
│       sub-MGH1_acq-uncombined08_T2starw.nii.gz
│       sub-MGH1_acq-uncombined09_T2starw.json
│       sub-MGH1_acq-uncombined09_T2starw.nii.gz
│       sub-MGH1_acq-uncombined10_T2starw.json
│       sub-MGH1_acq-uncombined10_T2starw.nii.gz
│       sub-MGH1_acq-uncombined11_T2starw.json
│       sub-MGH1_acq-uncombined11_T2starw.nii.gz
│       sub-MGH1_acq-uncombined12_T2starw.json
│       sub-MGH1_acq-uncombined12_T2starw.nii.gz
│       sub-MGH1_acq-uncombined13_T2starw.json
│       sub-MGH1_acq-uncombined13_T2starw.nii.gz
│       sub-MGH1_acq-uncombined14_T2starw.json
│       sub-MGH1_acq-uncombined14_T2starw.nii.gz
│       sub-MGH1_acq-uncombined15_T2starw.json
│       sub-MGH1_acq-uncombined15_T2starw.nii.gz
│       sub-MGH1_acq-uncombined16_T2starw.json
│       sub-MGH1_acq-uncombined16_T2starw.nii.gz
│       sub-MGH1_acq-uncombined17_T2starw.json
│       sub-MGH1_acq-uncombined17_T2starw.nii.gz
│       sub-MGH1_acq-uncombined18_T2starw.json
│       sub-MGH1_acq-uncombined18_T2starw.nii.gz
│       sub-MGH1_acq-uncombined19_T2starw.json
│       sub-MGH1_acq-uncombined19_T2starw.nii.gz
│       sub-MGH1_acq-uncombined20_T2starw.json
│       sub-MGH1_acq-uncombined20_T2starw.nii.gz
│       sub-MGH1_inv-1_MP2RAGE.json
│       sub-MGH1_inv-1_part-mag_MP2RAGE.nii.gz
│       sub-MGH1_inv-2_MP2RAGE.json
│       sub-MGH1_inv-2_part-mag_MP2RAGE.nii.gz
│       sub-MGH1_UNIT1_MP2RAGE.json
│       sub-MGH1_UNIT1_MP2RAGE.nii.gz
│
├───fmap
│       sub-MGH1_acq-coilQaSagLarge_SNR.json
│       sub-MGH1_acq-coilQaSagLarge_SNR.nii.gz
│       sub-MGH1_acq-coilQaSagSmall_GFactor.json
│       sub-MGH1_acq-coilQaSagSmall_GFactor.nii.gz
│       sub-MGH1_acq-coilQaTra_GFactor.json
│       sub-MGH1_acq-coilQaTra_GFactor.nii.gz
│       sub-MGH1_acq-dreamMedium066_e1_39.json
│       sub-MGH1_acq-dreamMedium066_e1_39.nii.gz
│       sub-MGH1_acq-dreamMedium066_e1_42.json
│       sub-MGH1_acq-dreamMedium066_e1_42.nii.gz
│       sub-MGH1_acq-dreamMedium066_e2_39.json
│       sub-MGH1_acq-dreamMedium066_e2_39.nii.gz
│       sub-MGH1_acq-dreamMedium066_e2_40.json
│       sub-MGH1_acq-dreamMedium066_e2_40.nii.gz
│       sub-MGH1_acq-dreamMedium066_e2_41.json
│       sub-MGH1_acq-dreamMedium066_e2_41.nii.gz
│       sub-MGH1_acq-dreamMedium066_e2_42.json
│       sub-MGH1_acq-dreamMedium066_e2_42.nii.gz
│       sub-MGH1_acq-dreamMedium066_e2_43.json
│       sub-MGH1_acq-dreamMedium066_e2_43.nii.gz
│       sub-MGH1_acq-dreamMedium066_e2_44.json
│       sub-MGH1_acq-dreamMedium066_e2_44.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e1_45.json
│       sub-MGH1_acq-dreamMediumHwlimit_e1_45.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e1_48.json
│       sub-MGH1_acq-dreamMediumHwlimit_e1_48.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e2_45.json
│       sub-MGH1_acq-dreamMediumHwlimit_e2_45.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e2_46.json
│       sub-MGH1_acq-dreamMediumHwlimit_e2_46.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e2_47.json
│       sub-MGH1_acq-dreamMediumHwlimit_e2_47.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e2_48.json
│       sub-MGH1_acq-dreamMediumHwlimit_e2_48.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e2_49.json
│       sub-MGH1_acq-dreamMediumHwlimit_e2_49.nii.gz
│       sub-MGH1_acq-dreamMediumHwlimit_e2_50.json
│       sub-MGH1_acq-dreamMediumHwlimit_e2_50.nii.gz
│       sub-MGH1_acq-dreamMedium_e1_33.json
│       sub-MGH1_acq-dreamMedium_e1_33.nii.gz
│       sub-MGH1_acq-dreamMedium_e1_36.json
│       sub-MGH1_acq-dreamMedium_e1_36.nii.gz
│       sub-MGH1_acq-dreamMedium_e2_33.json
│       sub-MGH1_acq-dreamMedium_e2_33.nii.gz
│       sub-MGH1_acq-dreamMedium_e2_34.json
│       sub-MGH1_acq-dreamMedium_e2_34.nii.gz
│       sub-MGH1_acq-dreamMedium_e2_35.json
│       sub-MGH1_acq-dreamMedium_e2_35.nii.gz
│       sub-MGH1_acq-dreamMedium_e2_36.json
│       sub-MGH1_acq-dreamMedium_e2_36.nii.gz
│       sub-MGH1_acq-dreamMedium_e2_37.json
│       sub-MGH1_acq-dreamMedium_e2_37.nii.gz
│       sub-MGH1_acq-dreamMedium_e2_38.json
│       sub-MGH1_acq-dreamMedium_e2_38.nii.gz
│       sub-MGH1_acq-famp_TB1TFL.json
│       sub-MGH1_acq-famp_TB1TFL.nii.gz
│       sub-MGH1_acq-mag_TB1TFL.json
│       sub-MGH1_acq-mag_TB1TFL.nii.gz
```

jcohenadad commented:
- `sub-MGH1_acq-uncombined01_T2starw.nii.gz` --> `sub-MGH1_rec-uncombined01_T2starw.nii.gz` also see definitions for https://bids-specification.readthedocs.io/en/stable/appendices/entities.html#rec and https://bids-specification.readthedocs.io/en/stable/appendices/entities.html#acq
- Where is the combined T2starw? It should also be present and named `sub-MGH1_T2starw.nii.gz`
- `sub-MGH1_UNIT1_MP2RAGE.nii.gz` --> `sub-MGH1_UNIT1.nii.gz` as per my comment https://github.com/spinal-cord-7t/coil-qc-code/pull/9#issuecomment-1909238742
- `sub-MGH1_acq-dreamMedium066_*` what does the `066` refer to? Maybe @evaalonsoortiz knows
- DREAM sequence, about the index numbers (_39, etc.): someone should look at the exact outputs of the DREAM sequence and name the appropriate outputs with either `_rec` or `_part`, depending on what these outputs are (also see https://bids-specification.readthedocs.io/en/stable/appendices/entities.html#part). Tagging @evaalonsoortiz
- `sub-MGH1_acq-mag_TB1TFL` Should be 'anat', see https://bids-specification.readthedocs.io/en/stable/appendices/qmri.html#tb1tfl-and-tb1rfm-specific-notes and https://openneuro.org/datasets/ds004906/versions/2.4.0

evaalonsoortiz commented:
> * Where is the combined T2starw? It should also be present and named `sub-MGH1_T2starw.nii.gz`

We had acquired the GRE scan in order to compare individual channels for each coil, therefore we are not interested in keeping the combined scan. 

> * `sub-MGH1_acq-dreamMedium066_*` what does the `066` refer to? Maybe @evaalonsoortiz knows

066 referes to the voltage. We acquired DREAM scans with 3 different voltages (0.66*refV, refV, 1.5*refV). The two contrasts output by the DREAM sequence (currently e1 and e2) can be combined into a FA map using https://github.com/spinal-cord-7t/coil-qc-code/blob/main/dream_FAmap.py. The scans acquired with 3 different voltages should also be combined into one scan to minimize bias in the FA map (see: https://pubmed.ncbi.nlm.nih.gov/32645432/). I will adapt [this script](https://github.com/spinal-cord-7t/coil-qc-code/blob/main/dream_FAmap.py) to do this. Once that is done, we can BIDSify that unbiased DREAM FA map only. 

> * DREAM sequence, about the index numbers (_39, etc.): someone should look at the exact outputs of the DREAM sequence and name the appropriate outputs with either `_rec` or `_part`, depending on what these outputs are (also see https://bids-specification.readthedocs.io/en/stable/appendices/entities.html#part). Tagging @evaalonsoortiz

Here my question is, what is the preferred convention for the following: 

We have 2 contrasts output by the DREAM scan, which are supposed to be combined to generate an FA map (see my comment above). 

Is it preferable to first combine and generate the FA map, then structure that according to BIDS (option 1), _OR_ is it preferred to have both contrasts in BIDS format, then leave it to the analysis script to grab those scans and generate an FA map (option 2)? 

It would be a similar process to generating a B0 map: is it preferred to BIDSify the individual echoes, or have a BIDS compliant B0 map in your open dataset? If there is no hard rule for this, I would prefer going with option 1 for our data.

jcohenadad commented:
> We had acquired the GRE scan in order to compare individual channels for each coil, therefore we are not interested in keeping the combined scan.

I consider the combined scan to be important, so I recommend keeping it. The uncombined and combined should be seen as a whole 'set' to qualitatively assess channel distributions and how the combined image looks in the end.

> 066 referes to the voltage. We acquired DREAM scans with 3 different voltages (0.66refV, refV, 1.5refV). 

In that case I would add the unit, eg: `acq-dreamMediumRefV0.66`, `acq-dreamMediumRefV1`, `acq-dreamMediumRefV1.5`

> Is it preferable to first combine and generate the FA map, then structure that according to BIDS (option 1), OR is it preferred to have both contrasts in BIDS format, then leave it to the analysis script to grab those scans and generate an FA map (option 2)?
It would be a similar process to generating a B0 map: is it preferred to BIDSify the individual echoes, or have a BIDS compliant B0 map in your open dataset? If there is no hard rule for this, I would prefer going with option 1 for our data.

There is no _hard rule_ for this, but my suggestion would be for option 2. Eg: if in 6 months we realize there was a bug in the way we compute the FA maps or the any other maps, then there is no "easy" going back because the source data would not be on OpenNeuro.

evaalonsoortiz commented:
@Kyota-exe Based on Julien's comments above, I will edit the script https://github.com/spinal-cord-7t/coil-qc-code/blob/main/dream_FAmap.py so that you have the code which computes an FA map from all of the DREAM scans. The code in that script can be transferred to the notebook that we put together for the full analysis.

Kyota-exe commented:
Understood, thank you @evaalonsoortiz.
@jcohenadad, thank you again for the feedback. I have implemented the latest filename modifications and added the corrected combined GRE scan to the database.
These are the contents of `sub-MGH1`:
```
├───anat
│       sub-MGH1_inv-1_MP2RAGE.json
│       sub-MGH1_inv-1_part-mag_MP2RAGE.nii.gz
│       sub-MGH1_inv-2_MP2RAGE.json
│       sub-MGH1_inv-2_part-mag_MP2RAGE.nii.gz
│       sub-MGH1_rec-uncombined01_T2starw.json
│       sub-MGH1_rec-uncombined01_T2starw.nii.gz
│       sub-MGH1_rec-uncombined02_T2starw.json
│       sub-MGH1_rec-uncombined02_T2starw.nii.gz
│       sub-MGH1_rec-uncombined03_T2starw.json
│       sub-MGH1_rec-uncombined03_T2starw.nii.gz
│       sub-MGH1_rec-uncombined04_T2starw.json
│       sub-MGH1_rec-uncombined04_T2starw.nii.gz
│       sub-MGH1_rec-uncombined05_T2starw.json
│       sub-MGH1_rec-uncombined05_T2starw.nii.gz
│       sub-MGH1_rec-uncombined06_T2starw.json
│       sub-MGH1_rec-uncombined06_T2starw.nii.gz
│       sub-MGH1_rec-uncombined07_T2starw.json
│       sub-MGH1_rec-uncombined07_T2starw.nii.gz
│       sub-MGH1_rec-uncombined08_T2starw.json
│       sub-MGH1_rec-uncombined08_T2starw.nii.gz
│       sub-MGH1_rec-uncombined09_T2starw.json
│       sub-MGH1_rec-uncombined09_T2starw.nii.gz
│       sub-MGH1_rec-uncombined10_T2starw.json
│       sub-MGH1_rec-uncombined10_T2starw.nii.gz
│       sub-MGH1_rec-uncombined11_T2starw.json
│       sub-MGH1_rec-uncombined11_T2starw.nii.gz
│       sub-MGH1_rec-uncombined12_T2starw.json
│       sub-MGH1_rec-uncombined12_T2starw.nii.gz
│       sub-MGH1_rec-uncombined13_T2starw.json
│       sub-MGH1_rec-uncombined13_T2starw.nii.gz
│       sub-MGH1_rec-uncombined14_T2starw.json
│       sub-MGH1_rec-uncombined14_T2starw.nii.gz
│       sub-MGH1_rec-uncombined15_T2starw.json
│       sub-MGH1_rec-uncombined15_T2starw.nii.gz
│       sub-MGH1_rec-uncombined16_T2starw.json
│       sub-MGH1_rec-uncombined16_T2starw.nii.gz
│       sub-MGH1_rec-uncombined17_T2starw.json
│       sub-MGH1_rec-uncombined17_T2starw.nii.gz
│       sub-MGH1_rec-uncombined18_T2starw.json
│       sub-MGH1_rec-uncombined18_T2starw.nii.gz
│       sub-MGH1_rec-uncombined19_T2starw.json
│       sub-MGH1_rec-uncombined19_T2starw.nii.gz
│       sub-MGH1_rec-uncombined20_T2starw.json
│       sub-MGH1_rec-uncombined20_T2starw.nii.gz
│       sub-MGH1_T2starw.json
│       sub-MGH1_T2starw.nii.gz
│       sub-MGH1_UNIT1.json
│       sub-MGH1_UNIT1.nii.gz
│
├───fmap
│       sub-MGH1_acq-anat_TB1TFL.json
│       sub-MGH1_acq-anat_TB1TFL.nii.gz
│       sub-MGH1_acq-coilQaSagLarge_SNR.json
│       sub-MGH1_acq-coilQaSagLarge_SNR.nii.gz
│       sub-MGH1_acq-coilQaSagSmall_GFactor.json
│       sub-MGH1_acq-coilQaSagSmall_GFactor.nii.gz
│       sub-MGH1_acq-coilQaTra_GFactor.json
│       sub-MGH1_acq-coilQaTra_GFactor.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e1-33.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e1-33.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e1-36.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e1-36.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-33.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-33.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-34.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-34.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-35.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-35.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-36.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-36.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-37.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-37.nii.gz
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-38.json
│       sub-MGH1_acq-dreamMediumRefV0.66-e2-38.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e1-39.json
│       sub-MGH1_acq-dreamMediumRefV1-e1-39.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e1-42.json
│       sub-MGH1_acq-dreamMediumRefV1-e1-42.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e2-39.json
│       sub-MGH1_acq-dreamMediumRefV1-e2-39.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e2-40.json
│       sub-MGH1_acq-dreamMediumRefV1-e2-40.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e2-41.json
│       sub-MGH1_acq-dreamMediumRefV1-e2-41.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e2-42.json
│       sub-MGH1_acq-dreamMediumRefV1-e2-42.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e2-43.json
│       sub-MGH1_acq-dreamMediumRefV1-e2-43.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1-e2-44.json
│       sub-MGH1_acq-dreamMediumRefV1-e2-44.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e1-45.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e1-45.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e1-48.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e1-48.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-45.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-45.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-46.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-46.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-47.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-47.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-48.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-48.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-49.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-49.nii.gz
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-50.json
│       sub-MGH1_acq-dreamMediumRefV1.5-e2-50.nii.gz
│       sub-MGH1_acq-famp_TB1TFL.json
│       sub-MGH1_acq-famp_TB1TFL.nii.gz
```

jcohenadad commented:
Almost there! Few additional things:
- Echoes need to be distinguished, see [BIDS website](https://bids-specification.readthedocs.io/en/stable/appendices/entities.html#echo). So: 
  `sub-MGH1_acq-dreamMediumRefV0.66-e1-33.json` --> `sub-MGH1_acq-dreamMediumRefV0.66_echo-1-33.json`
- The suffix "33" has not been addressed. We cannot use the run number as the suffix. One of the reason we do this BIDS conversion is to be able to use software that runs on a bunch of files that assume standardized file names. If, for example, the analysis pipeline requires that, let's say, the phase of echo 1 of the DREAM sequence needs to be processed, then how will we be able to distinguish it if the 'phase' information is not in the file name? So, as I said [earlier](https://github.com/spinal-cord-7t/coil-qc-code/pull/9#issuecomment-1913180665), can you or @evaalonsoortiz explicit what do '33', '34', etc. of the DREAM sequence refer to? Once identified, we need to ensure that the ordering of the index is systematic across multiple subjects, and once confirmed, we can systematize the data conversion with code

evaalonsoortiz commented:
The DREAM scans listed above correspond to the following:

e1-33 : non-distortion corrected (ND) STE image
e1-36: distortion corrected (DIS2D) STE image

e2-33 : ND FID image
e2-36: DIS2D FID image

e2-34 : ND reference voltage (refV) map
e2-37: DIS2D refV map

e2-35: ND flip angle (FA) map
e2-38: DIS2D FA map

STE and FID images can be combined to create an FA map (see: https://github.com/spinal-cord-7t/coil-qc-code/blob/main/dream_FAmap.py, and https://onlinelibrary.wiley.com/doi/10.1002/mrm.24158). Apparently both FA and voltage maps are already output by the scanner. I suggest we keep only the FA and RefV maps, and not STE and FID images. TBD how the BIDS convention deals with those designations (FA and RefV maps). 

Now we have to choose between ND and DIS2D. I have looked at images for the dataset corresponding to MGH1 and it appears that MP2RAGE was acquired with no distortion correction (even though the protocol circulated by Virginie had distortion correction ON). 
- TFLflash was acquired with distortion correction set to ON, but the json states that the images are ND. 
- DREAM was acquired with distortion correction set to ON, and the json files indicate that there is one set of images with ND and one with DIS2D
- coilQA images are all ND as intended

Conclusion: 
- Distortion correction for TFLflash perhaps does nothing (as was the case for coilQA, which is why we never turned it on)
- We should keep the ND images for DREAM and not the DIS2D, since we have no other DIS2D scans (not is the MP2RAGE DIS2D, and that is the one we use for segmenting the cord).
- I also looked at the ND vs. DIS2D DREAM images and the distortion effect is minimal

evaalonsoortiz commented:
Yes, we should get rid of e1, e2.

Kyota-exe commented:
I have modified the conversion script in accordance with the discussion, these are the latest outputs:
```
├───anat
│       sub-MGH1_inv-1_MP2RAGE.json
│       sub-MGH1_inv-1_part-mag_MP2RAGE.nii.gz
│       sub-MGH1_inv-2_MP2RAGE.json
│       sub-MGH1_inv-2_part-mag_MP2RAGE.nii.gz
│       sub-MGH1_rec-uncombined01_T2starw.json
│       sub-MGH1_rec-uncombined01_T2starw.nii.gz
│       sub-MGH1_rec-uncombined02_T2starw.json
│       sub-MGH1_rec-uncombined02_T2starw.nii.gz
│       sub-MGH1_rec-uncombined03_T2starw.json
│       sub-MGH1_rec-uncombined03_T2starw.nii.gz
│       sub-MGH1_rec-uncombined04_T2starw.json
│       sub-MGH1_rec-uncombined04_T2starw.nii.gz
│       sub-MGH1_rec-uncombined05_T2starw.json
│       sub-MGH1_rec-uncombined05_T2starw.nii.gz
│       sub-MGH1_rec-uncombined06_T2starw.json
│       sub-MGH1_rec-uncombined06_T2starw.nii.gz
│       sub-MGH1_rec-uncombined07_T2starw.json
│       sub-MGH1_rec-uncombined07_T2starw.nii.gz
│       sub-MGH1_rec-uncombined08_T2starw.json
│       sub-MGH1_rec-uncombined08_T2starw.nii.gz
│       sub-MGH1_rec-uncombined09_T2starw.json
│       sub-MGH1_rec-uncombined09_T2starw.nii.gz
│       sub-MGH1_rec-uncombined10_T2starw.json
│       sub-MGH1_rec-uncombined10_T2starw.nii.gz
│       sub-MGH1_rec-uncombined11_T2starw.json
│       sub-MGH1_rec-uncombined11_T2starw.nii.gz
│       sub-MGH1_rec-uncombined12_T2starw.json
│       sub-MGH1_rec-uncombined12_T2starw.nii.gz
│       sub-MGH1_rec-uncombined13_T2starw.json
│       sub-MGH1_rec-uncombined13_T2starw.nii.gz
│       sub-MGH1_rec-uncombined14_T2starw.json
│       sub-MGH1_rec-uncombined14_T2starw.nii.gz
│       sub-MGH1_rec-uncombined15_T2starw.json
│       sub-MGH1_rec-uncombined15_T2starw.nii.gz
│       sub-MGH1_rec-uncombined16_T2starw.json
│       sub-MGH1_rec-uncombined16_T2starw.nii.gz
│       sub-MGH1_rec-uncombined17_T2starw.json
│       sub-MGH1_rec-uncombined17_T2starw.nii.gz
│       sub-MGH1_rec-uncombined18_T2starw.json
│       sub-MGH1_rec-uncombined18_T2starw.nii.gz
│       sub-MGH1_rec-uncombined19_T2starw.json
│       sub-MGH1_rec-uncombined19_T2starw.nii.gz
│       sub-MGH1_rec-uncombined20_T2starw.json
│       sub-MGH1_rec-uncombined20_T2starw.nii.gz
│       sub-MGH1_T2starw.json
│       sub-MGH1_T2starw.nii.gz
│       sub-MGH1_UNIT1.json
│       sub-MGH1_UNIT1.nii.gz
│
├───fmap
│       sub-MGH1_acq-anat_TB1TFL.json
│       sub-MGH1_acq-anat_TB1TFL.nii.gz
│       sub-MGH1_acq-coilQaSagLarge_SNR.json
│       sub-MGH1_acq-coilQaSagLarge_SNR.nii.gz
│       sub-MGH1_acq-coilQaSagSmall_GFactor.json
│       sub-MGH1_acq-coilQaSagSmall_GFactor.nii.gz
│       sub-MGH1_acq-coilQaTra_GFactor.json
│       sub-MGH1_acq-coilQaTra_GFactor.nii.gz
│       sub-MGH1_acq-famp_TB1DREAM.json
│       sub-MGH1_acq-famp_TB1DREAM.nii.gz
│       sub-MGH1_acq-famp_TB1TFL.json
│       sub-MGH1_acq-famp_TB1TFL.nii.gz
│       sub-MGH1_acq-refv_TB1DREAM.json
│       sub-MGH1_acq-refv_TB1DREAM.nii.gz
```
I will make sure the conversion script works correctly for all the subjects in every site, but unless I missed something, I believe the conversion should be good now. :smile:

jcohenadad commented:
Amazing! Good job @Kyota-exe 👏

evaalonsoortiz commented:
@Kyota-exe I think we may have forgotten to convert the DREAM 0.66*refV and 1.5*refV DREAM scans? They are not included in the dataset you've shared on gdrive.
